### PR TITLE
Randomize last 2 bytes(5th and 6th) of MAC address in node policy.

### PR DESCRIPTION
**Pull request by sudhar-krishnakumar**
Randomize last 2 bytes(5th and 6th) of MAC address in node policy.

### CODE REVIEW

File: ipu-plugin/pkg/ipuplugin/lifecycleservice.go [hunk 1]
```diff
@@ -46,12 +47,13 @@ type LifeCycleServiceServer struct {
...
-	vendorId    = "0x8086"
-	imcAddress  = "192.168.0.1:22"
-	apfNumber   = 16
+	hostVportId         = "03"
+	accVportId          = "04"
+	deviceId            = "0x1452"
+	vendorId            = "0x8086"
+	imcAddress          = "192.168.0.1:22"
+	apfNumber           = 16
+	last_byte_mac_range = 246
```
Comment (vkannapp) [inline ipu-plugin/pkg/ipuplugin/lifecycleservice.go:56]: https://github.com/intel-innersource/drivers.ethernet.linux.mev-cp/blob/7edcb25af59a930de8e932ba095bd8668e765b00/src/mac_utils.c#L181

ipu_platform/imc/agents/userspace/mev_imc_dpcp_src/src/libdpcp/src/mac_utils.c

81:  base_mac_addr[5] = policy mac_addr[5] + func_id

The 6th byte of base mac addr is set for all 16 function ids and the randomization should be in slot of 16 mac addresses.

last_byte_mac_range should be in the range of 256-16 = 240.

File: ipu-plugin/pkg/ipuplugin/lifecycleservice.go [hunk 2]
```diff
@@ -333,6 +335,27 @@ func configureChannel(mode, daemonHostIp, daemonIpuIp string) error {
...
+// sets random bytes for last 2 bytes(5th and 6th) in MAC address
+func setBaseMacAddr() (string, error) {
+	var macAddress string
+	macBytes := make([]byte, 2)
+	_, err := rand.Read(macBytes)
+	if err != nil {
+		return "", fmt.Errorf("error->%v, failed to create random bytes for MAC: ", err)
+	}
+	//Restricting range of last byte in node-policy to be less than 247, to
+	//allow for last byte of MAC+4 for ACC, and last byte of MAC+8 for IMC, as observed in cli-client output.
```
Comment (vkannapp) [inline ipu-plugin/pkg/ipuplugin/lifecycleservice.go:347]: Last byte is derived as base_mac_addr[5] = policy mac_addr[5] + func_id

ACC: policy_mac_addr[5] + 4
IMC: policy mac_addr[5] + 5

**Review by vkannapp — APPROVED** [review]
Great work Sudhar! Approved the updated commits.

**Comment by sudhar-krishnakumar** [comment]
Hi Sal, @SalDaniele , if possible would like your inputs, once you are able to test this change in cluster. Branch is here-> https://github.com/sudhar-krishnakumar/ipu-opi-plugins/tree/update_mac

**Pull request merged**